dashboard/app/models/foorm/form.rb

```diff
@@ -84,4 +90,59 @@ def self.fill_in_library_items(questions)
     end
     return questions
   end
+
+  def self.get_matrix_question_id(parent_question_id, sub_question_id)
+    parent_question_id + '-' + sub_question_id
+  end
+
+  # For a given Form, this method will produce a CSV of all responses
+  # received for that Form. It includes the content of the form submitted
+  # by a user, as well as some additional metadata about the context
+  # in which the form was submitted (eg, workshop ID, user ID).
+  def submissions_to_csv(file_path)
+    formatted_submissions = []
+
+    submissions.each do |submission|
+      formatted_submission = submission.formatted_answers
+      formatted_submission.merge! submission.formatted_metadata
+
+      formatted_submissions << formatted_submission
+    end
+
+    CSV.open(file_path, "wb") do |csv|
+      break if formatted_submissions.empty?
+
+      headers = readable_questions
+      headers_with_config_variables = formatted_submissions.first.keys.map do |question_id|
+        headers[question_id] || question_id
+      end
+
+      csv << headers_with_config_variables
+      formatted_submissions.each do |submission|
+        csv << submission.values_at(*headers_with_config_variables)
+      end
+    end
+  end
+
+  def readable_questions
+    questions = {}
+
+    parsed_questions[:general][key].each do |question_id, question_details|
+      if question_details[:type] == 'matrix'
+        matrix_questions = question_details[:rows]
+        matrix_questions.each do |matrix_question_id, matrix_question_text|
+          key = self.class.get_matrix_question_id(question_id, matrix_question_id)
+          questions[key] = question_details[:title] + ' >> ' + matrix_question_text
```

Comment:
I think we should check with Baker/whoever else is interested in this csv to see what they want the column names to be. The column names end up extremely long with this formatting, so keys may be a better option since they are human-readable anyway

Reply:
I checked, he said roll with long column names. I think majority use case here is people wanting a CSV download they can quickly look at in gsheets, where I think you'd want to have the full question visible.